mitodl/bootcamp-ecommerce
# Fix hubspot sync issues

#### What are the relevant tickets?
Closes #679 

#### What's this PR do?
- Adds a retry to the application syncing task.
- The application signal only triggers on update instead of create, because the `applications.api.get_or_create_bootcamp_application` function always saves again after a create
- Adds a signal for bootcamps so they will be synced with hubspot as new products when created from django admin
- Updates the hubspot error checking task, which was overlooked last time, and adds more tests for it

#### How should this be manually tested?
- Create and apply for a bunch of bootcamps on RC, within a few minutes they should all be on hubspot
- Add a new bootcamp in Django admin, should be on heroku within a few minutes (can get the product list at https://app.hubspot.com/settings/6431386/sales/products)
- Tests should pass
